From cbc82816a87fcce27c4f5258b6988306636ed1dc Mon Sep 17 00:00:00 2001
From: SeungYoon Lee <w22905@motorola.com>
Date: Tue, 7 Oct 2014 00:01:04 -0500
Subject: [PATCH 083/959] IKVPREL1L-615 driver: misc: stml0xx: Add headset
 report capabilities

Add SW_HEADPHONE_INSERT and SW_MICROPHONE_INSERT capabilities
to report headset and headset type accessories insertion.
Add KEY_MEDIA to handle default action which is common in KK and L.

Change-Id: Ib0c39ed5f55cf609eab5e3c1121d87a304a39156
Signed-off-by: SeungYoon Lee <w22905@motorola.com>
Reviewed-on: http://gerrit.mot.com/676719
SLTApproved: Slta Waiver <sltawvr@motorola.com>
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Robert Wiegand <Robert.P.Wiegand@motorola.com>
Reviewed-by: Vladimir Karpovich <vkarpovich@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
---
 drivers/misc/stml0xx/stml0xx_core.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/misc/stml0xx/stml0xx_core.c b/drivers/misc/stml0xx/stml0xx_core.c
index 8f1e685ca05..0b67d7c2635 100644
--- a/drivers/misc/stml0xx/stml0xx_core.c
+++ b/drivers/misc/stml0xx/stml0xx_core.c
@@ -922,7 +922,10 @@ static int stml0xx_probe(struct spi_device *spi)
 	input_set_drvdata(ps_stml0xx->input_dev, ps_stml0xx);
 	input_set_capability(ps_stml0xx->input_dev, EV_KEY, KEY_POWER);
 	input_set_capability(ps_stml0xx->input_dev, EV_KEY, KEY_CAMERA);
+	input_set_capability(ps_stml0xx->input_dev, EV_KEY, KEY_MEDIA);
 	input_set_capability(ps_stml0xx->input_dev, EV_SW, SW_LID);
+	input_set_capability(ps_stml0xx->input_dev, EV_SW, SW_HEADPHONE_INSERT);
+	input_set_capability(ps_stml0xx->input_dev, EV_SW, SW_MICROPHONE_INSERT);
 	ps_stml0xx->input_dev->name = "sensorprocessor";
 
 	err = input_register_device(ps_stml0xx->input_dev);
-- 
2.11.0

